mesheryctl/internal/cli/root/system/token_test.go

```diff
@@ -0,0 +1,426 @@
+		},
+		{
+			Name:                 "delete the passed token(with a token name that doesn't exist)",
+			Args:                 []string{"token", "create", "Default2"},
+			ExpectedResponse:     "delete.golden",
+			ExpectedResponseYaml: "delete_err.yaml",
+			ExpectErr:            true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			utils.SetupCustomContextEnv(t, currDir+"/testdata/token/"+tt.ExpectedResponseYaml)
+			var b *bytes.Buffer
+			var e *bytes.Buffer
+			if tt.ExpectErr {
+				e = utils.SetupLogrusGrabTesting(t)
+				SystemCmd.SetErr(e)
+			} else {
+				b = utils.SetupLogrusGrabTesting(t)
+				SystemCmd.SetOut(b)
+			}
+			SystemCmd.SetArgs(tt.Args)
+			var actualResponse string
+			err := SystemCmd.Execute()
+			if err != nil {
+				if !tt.ExpectErr {
+					t.Error(err)
+					return
+				}
+			}
+			//Check the stdout/stderr against the golden file
+			if tt.ExpectErr {
+				actualResponse = e.String()
+			} else {
+				actualResponse = b.String()
+			}
+			// Expected response
+			testdatatokenDir := filepath.Join(currDir, "testdata/token")
+			golden := utils.NewGoldenFile(t, tt.ExpectedResponse, testdatatokenDir)
+			if *update {
+				golden.Write(actualResponse)
+			}
+			expectedResponse := golden.Load()
+			if expectedResponse != actualResponse {
+				t.Errorf("expected response [%v] and actual response [%v] don't match", expectedResponse, actualResponse)
+			}
+			//Skip checking the yamls if we had an error
+			if tt.ExpectErr {
+				return
+			}
+			//Check the modified yaml against the golden file
+			path, err := os.Getwd()
+			if err != nil {
+				t.Error("unable to locate meshery directory")
+			}
+			filepath := path + "/testdata/token/" + tt.ExpectedResponseYaml
+			content, err := ioutil.ReadFile(filepath)
+			if err != nil {
+				t.Error(err)
+			}
+			actualResponse = string(content)
+			golden = utils.NewGoldenFile(t, tt.ExpectedResponseYaml, testdatatokenDir)
+			if *update {
+				golden.Write(actualResponse)
+			}
+			createExpected := golden.Load()
+			if actualResponse != createExpected {
+				t.Errorf("expected response %v and actual response %v don't match", createExpected, actualResponse)
+			}
+			if err := utils.Copy(path+"/fixtures/.meshery/config.yaml", filepath); err != nil {
+				t.Error(err, "Could not complete test. Unable to configure create test file")
+			}
+			BreakupFunc(t)
+		})
+	}
+}
+
+func TestTokenSetCmd(t *testing.T) {
+	// get current directory
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("Not able to get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	tests := []utils.CmdTestInput{
+		{
+			Name:                 "set the token for default context",
+			Args:                 []string{"token", "set", "DefaultNew"},
+			ExpectedResponse:     "set_default.golden",
+			ExpectedResponseYaml: "set_default.yaml",
+			ExpectErr:            false,
+		},
+		{
+			Name:                 "set the token for custom context",
```

Comment:
```suggestion
			Name:                 "set the token for different context",
```
plus one case what if that context doesn't exist!